src/platform/plugins/shared/dashboard/public/dashboard_api/unified_search_manager.test.ts

```diff
@@ -103,34 +103,34 @@ describe('initializeUnifiedSearchManager', () => {
             useUnifiedSearchIntegration: false,
           }
         );
-        let unsavedChanges: object | undefined;
+        let emitCount = 0;
         unifiedSearchManager.internalApi.startComparing$(lastSavedState$).subscribe((changes) => {
```

Comment:
It might be good to use a `take(2)` here to ensure unsubscription, and show explicitly that you're looking for two emits only.